### PR TITLE
TST: check inequality by comparing categorical with NaN ( #28384 )

### LINKED ISSUE #28384: Categorical equality with NaN behaves unexpectedly
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd

cat_type = pd.CategoricalDtype(categories=["a", "b", "c"])
s1_cat = pd.Series(["a", "b", "c"], dtype=cat_type)
s2_cat = pd.Series([np.nan, "a", "b"], dtype=cat_type)
s1_cat != s2_cat  # expected all True
```
Output:
```
0    False
1     True
2     True
dtype: bool
```

#### Problem description
I would expect anything to compare `!=` to NaN.

Comparing the categorical series with `==` works as expected:
```python
s1_cat == s2_cat  # expect all False
```
Output:
```
0    False
1    False
2    False
dtype: bool
```

Element-wise comparison seems to work fine:
```python
for left, right in zip(s1_cat, s2_cat):
    print(left != right)  # expect all True
```
Output:
```
True
True
True
```

This also works as expected with other data types, e.g.:
```python
s1_int = pd.Series([1, 2, 3])
s2_int = pd.Series([np.nan, 1, 2])
s1_int != s2_int  # expect all True
```
gives
```
0    True
1    True
2    True
dtype: bool
```

Apologies if this is a duplicate; I have found various issues about categorical types and missing values, but not about comparing them in this way.

#### Expected Output
```python
s1_cat != s2_cat  # expected all True
```

```
0    True
1    True
2    True
dtype: bool
```
#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.4.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 16.7.0
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : en_GB.UTF-8
LOCALE           : en_GB.UTF-8

pandas           : 0.25.1
numpy            : 1.17.2
pytz             : 2019.2
dateutil         : 2.8.0
pip              : 19.2.2
setuptools       : 41.0.1
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None

</details>

### PR DESCRIPTION
**Pull request by junjunjunk**
- [x] closes #28384
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by junjunjunk** [comment]
Please review.

And I need some help.  
I don't know why CI is failing.From the looks of it, it looks like it's timed out.

File: pandas/tests/arrays/categorical/test_missing.py [hunk 1]
```diff
@@ -148,3 +148,13 @@ def test_use_inf_as_na_outside_context(self, values, expected):
             result = pd.isna(DataFrame(cat))
             expected = DataFrame(expected)
             tm.assert_frame_equal(result, expected)
+
+    def test_compare_unequal(self):
```
Comment (arw2019) [inline pandas/tests/arrays/categorical/test_missing.py:152]: `test_compare_categorical_with_missing`

**Review by arw2019 — COMMENTED** [review]
I think possibly add a non-categorical example since the OP was about the difference in behavior between categorical vs. non-categorical. See https://github.com/pandas-dev/pandas/issues/28384#issuecomment-530514218

Re: CI failures those are unrelated I think. Merge with master and they'll be rerun

File: pandas/tests/arrays/categorical/test_missing.py (hunk 1 above)
Comment (junjunjunk) [reply pandas/tests/arrays/categorical/test_missing.py]: Thank you. Fixed.
https://github.com/pandas-dev/pandas/pull/36520/commits/3e293cf0d223322d81a5f6ac632f15973b64335e

**Comment by junjunjunk** [comment]
> I think possibly add a non-categorical example since the OP was about the difference in behavior between categorical vs. non-categorical. See [#28384 (comment)](https://github.com/pandas-dev/pandas/issues/28384#issuecomment-530514218)
> 
> Re: CI failures those are unrelated I think. Merge with master and they'll be rerun

Thank you for your input. I added test case.
And I rebased my branch.

File: pandas/tests/arrays/categorical/test_missing.py [hunk 2]
```diff
@@ -148,3 +148,22 @@ def test_use_inf_as_na_outside_context(self, values, expected):
             result = pd.isna(DataFrame(cat))
             expected = DataFrame(expected)
             tm.assert_frame_equal(result, expected)
+
+    def test_compare_categorical_with_missing(self):
```
Comment (jreback) [inline pandas/tests/arrays/categorical/test_missing.py:152]: can you fully replicate the tests in the OP, e.g. add for == as well.

ideally parameterize this over these 4 cases (or 2 and 2 in-line if easier).

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/arrays/categorical/test_missing.py (hunk 2 above)
Comment (junjunjunk) [reply pandas/tests/arrays/categorical/test_missing.py]: Thank you for review. Done.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @junjunjunk

**Pull request merged**